compiler/rustc_trait_selection/src/traits/on_unimplemented.rs

```diff
@@ -185,7 +184,7 @@ impl<'tcx> OnUnimplementedDirective {
                 enclosing_scope: None,
             }))
         } else {
-            return Err(ErrorReported);
+            return Err(ErrorReported::new_i_know_what_im_doing());
```

Comment:
This probably needs to be `delay_span_bug`, the presence of a past error is merely implied here.